src/main/java/subway/application/SectionService.java

```diff
@@ -0,0 +1,82 @@
+package subway.application;
+
+import org.springframework.stereotype.Service;
+import subway.dao.SectionDao;
+import subway.domain.Section;
+import subway.domain.Sections;
+import subway.dto.SectionRequest;
+import subway.dto.SectionResponse;
+import subway.exception.SubwayException;
+
+@Service
+public class SectionService {
+
+    private final SectionDao sectionDao;
+
+    public SectionService(final SectionDao sectionDao) {
+        this.sectionDao = sectionDao;
+    }
+
+    public SectionResponse createSection(final Long lineId, final SectionRequest sectionRequest) {
+        final Sections sections = new Sections(sectionDao.findAllByLineId(lineId));
+        validateNotContainsBothStation(sections, sectionRequest);
+
+        if (sections.isEndStation(sectionRequest.getUpStationId(), sectionRequest.getDownStationId())) {
+            final Section section = sectionDao.insert(sectionRequest.toSection(lineId));
+            return SectionResponse.of(section);
+        }
+
+        return SectionResponse.of(insertBetween(sections, sectionRequest));
+    }
+
+    private Section insertBetween(final Sections sections, final SectionRequest sectionRequest) {
+        final Section targetSection =
+                sections.findContainStationSection(sectionRequest.getUpStationId(), sectionRequest.getDownStationId())
+                        .orElseThrow(() -> new SubwayException("상행 역과 하행 역이 모두 노선에 없습니다."));
+
+        validateDistance(targetSection.subtractDistance(sectionRequest.getDistance()));
+        final Section requestSection = sectionRequest.toSection(targetSection.getLineId());
+        final Section remainSection = targetSection.subtract(requestSection);
+
+        final Section result = sectionDao.insert(requestSection);
+        sectionDao.insert(remainSection);
+        sectionDao.delete(targetSection.getId());
+
+        return result;
+    }
+
+    public void deleteSection(final Long lineId, final Long downStationId) {
+        final Sections sections = new Sections(sectionDao.findAllByLineId(lineId));
+        validateSectionsSizeIsNotOne(sections);
+
+        final Section lastSection = getLastSection(sections);
+        if (!lastSection.isSameDownStationId(downStationId)) {
+            throw new SubwayException("해당 노선에 일치하는 하행 종점역이 존재하지 않습니다.");
```

Comment:
오류 상황에 대한 메시지를 좀더 풍부하게 만들면 미래에 자신에게 감사하게 됩니다.
```suggestion
            throw new SubwayException("해당 노선에 일치하는 하행 종점역이 존재하지 않습니다. 노선 ID : " + lineId + ", 하행 좀점역 ID : " + downStationId);
```

Reply:
좋은 것 같아요! 이렇게 하면 나중에 오류가 발생했을 때 좀 더 디테일하게 알 수 있을 것 같아요!